offset.go

```diff
@@ -0,0 +1,96 @@
+package vmap
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+)
+
+type (
+	// Duration is a time.Duration alias
+	Duration time.Duration
+
+	// Offset is a VAMP duration expressed as  hh:mm:ss
+	Offset struct {
+		timeDur  Duration
+		position OffsetPosition
+		percent  int
+	}
+
+	// OffsetPosition is for either passing start/end as the position of the ad
+	OffsetPosition string
+)
+
+const (
+	// OffsetPositionStart "start"
+	OffsetPositionStart OffsetPosition = "start"
+	// OffsetPositionEnd "end"
+	OffsetPositionEnd OffsetPosition = "end"
+)
+
+// MarshalText implements the encoding.TextMarshaler interface.
+func (off *Offset) MarshalText() ([]byte, error) {
+	if off.position == OffsetPositionStart || off.position == OffsetPositionEnd {
+		return []byte(off.position), nil
+	}
+
+	if off.percent != 0 {
+		return []byte(fmt.Sprintf("%d%%", off.percent)), nil
+	}
+
+	dur := off.timeDur
+	h := dur / Duration(time.Hour)
+	m := dur % Duration(time.Hour) / Duration(time.Minute)
```

Comment:
aren't you supposed to subtract one from another or it's only purpose is to demonstrate the conversion to different time entities ?

Reply:
i dont think so. i got this from our VAST repo. the modulus operate basically removes anything larger than an hour and that just leaves the minutes to get counted